samples/Samples.RateLimiter/Program.cs

```diff
@@ -0,0 +1,131 @@
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Reflection;
+using System.Threading;
+using Datadog.Trace;
+using Datadog.Trace.Configuration;
+
+namespace Samples.RateLimiter
+{
+    internal static class Program
+    {
+        private static readonly string ServiceDogWalker = "dog_walker";
+        private static readonly string RootWalkOperation = "root_walk";
+
+        private static readonly Dictionary<string, int> Counts = new Dictionary<string, int>();
+
+        private static void Main()
+        {
+            var numberOfSeconds = 5;
+            var maxMilliseconds = numberOfSeconds * 1000;
+            var configuredLimitPerSecond = int.Parse(Environment.GetEnvironmentVariables()[ConfigurationKeys.MaxTracesSubmittedPerSecond].ToString());
+
+            Console.WriteLine($"Ready to run for {numberOfSeconds} seconds.");
+            Console.WriteLine($"Configured rate limit of {configuredLimitPerSecond}");
+
+            PrepKeys(ServiceDogWalker, RootWalkOperation, configuredLimitPerSecond * numberOfSeconds);
+
+            var timer = new Stopwatch();
+
+            timer.Start();
+
+            while (true)
+            {
+                if (timer.ElapsedMilliseconds >= maxMilliseconds)
+                {
+                    timer.Stop();
+                    break;
+                }
+
+                RunStuff(ServiceDogWalker, RootWalkOperation);
+            }
+
+            Console.WriteLine();
+
+            foreach (var key in Counts.Keys)
+            {
+                var isExpect = key.Contains("Expecting");
+
+                if (isExpect)
+                {
+                    Console.WriteLine();
+                    Console.WriteLine($"{key}");
+                }
+                else
+                {
+                    Console.WriteLine($"{key}: {Counts[key]}");
+                }
+            }
+        }
+
+        private static void RunStuff(string serviceName, string operationName)
+        {
+            Console.Write(".");
+
+            Counts[Key(serviceName, operationName)]++;
+
+            Scope root;
+
+            using (root = Tracer.Instance.StartActive(operationName: operationName, serviceName: serviceName))
```

Comment:
Tests that only use manual instrumentation don't need an additional project, since we don't need to attach the profiler.